server/datastore/mysql/migrations/tables/20240221112844_FixUniquePolicyNameBug.go

```diff
@@ -0,0 +1,107 @@
+package tables
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"github.com/VividCortex/mysqlerr"
+	"github.com/go-sql-driver/mysql"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20240221112844, Down_20240221112844)
+}
+
+func Up_20240221112844(tx *sql.Tx) error {
+	// The bug was that the checksum was not updating. So we need to update the checksum for existing rows.
+	const updateStmt = `
+	UPDATE
+		policies
+	SET
+		checksum = UNHEX(
+			MD5(
+				-- concatenate with separator \x00
+				CONCAT_WS(CHAR(0),
+					COALESCE(team_id, ''),
+					name
+				)
+			)
+		)
+	`
+	const updateNameStmt = `
+	UPDATE
+		policies
+	SET
+		name = CONCAT(name, '%d'),
+		checksum = UNHEX(
+			MD5(
+				-- concatenate with separator \x00
+				CONCAT_WS(CHAR(0),
+					COALESCE(team_id, ''),
+					name
+				)
+			)
+		)
+	WHERE id = ?
+	`
+	_, err := tx.Exec(updateStmt)
+	isDuplicate := func(err error) bool {
+		var driverErr *mysql.MySQLError
+		if errors.As(err, &driverErr) && driverErr.Number == mysqlerr.ER_DUP_ENTRY {
+			return true
+		}
+		return false
+	}
+
+	duplicateExists := false
+	if err != nil {
+		if isDuplicate(err) {
+			duplicateExists = true
+		} else {
+			return fmt.Errorf("failed to update policies table to fill the checksum column: %w", err)
+		}
+	}
+
+	// Since there is a duplicate name somewhere in the table (should be rare), we update table 1 row at a time
+	if duplicateExists {
+		rows, err := tx.Query("SELECT id FROM policies")
+		if err != nil {
+			return fmt.Errorf("failed to query policies table: %w", err)
+		} else if rows.Err() != nil {
+			return fmt.Errorf("failed to query policies table: %w", rows.Err())
+		}
+		var ids []uint64
+		{
```

Comment:
Curious as to why the scoping?

Reply:
Solved by using an anonymous inline function:
```go
func() {
  ...
}()
```

Reply:
removed extra scope